### PR TITLE
Background the Launch Queue

### PR DESCRIPTION
**Pull request by Meatballs1**
This is an example of queueing the Launch command so that we do not have to worry about the page timeout for even really large jobs.

### CODE REVIEW

**Comment by Meatballs1** [comment]
The diff between 61d2f0e and 1d4248f shows a single background task versus a single task spawning additional tasks...

**Comment by Meatballs1** [comment]
I think the approach of the background process spawning additional background processes is probably a bit nicer. We can stop sidekiq at a pinch if we want to stop mailings etc.

It also means we could potentially take advantage of sidekiq retries in future if we wanted to for individual emails.

File: app/workers/queue_mail_worker.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    if campaign.errors.present?
+      logger.info "Unable to launch campaign #{campaign_id} as errors are present."
+      return false
+    end
+    campaign.update_attributes(active: true, email_sent: true)
+    blast = campaign.blasts.create(test: false)
+    victims = Victim.where("campaign_id = ? and archive = ?", campaign_id, false)
+    logger.info "Queueing #{victims.count} emails for background delivery for campaign #{campaign_id}"
+    victims.each do |target|
+      MailWorker.perform_async(campaign.id, target.email_address, blast.id, PhishingFrenzyMailer::ACTIVE)
```
Comment (zeknox) [inline app/workers/queue_mail_worker.rb]: What am I missing here?  Isn't there suppose to be a mail_worker.rb file?

**Comment by zeknox** [comment]
Nice work @Meatballs1, super excited about this one!

File: app/workers/queue_mail_worker.rb (hunk 1 above)
Comment (Meatballs1) [inline app/workers/queue_mail_worker.rb]: Yeah it uses the existing mail worker?
________________________________

Comment (Meatballs1) [inline app/workers/queue_mail_worker.rb]: app/workers/mail_worker.rb

**Pull request merged**